File: training/src/classesandobjects/scala/Customer3.scala
package classesandobjects.scala

class Customer3(private var _id: String, val name: String, val address: String, x: String) {
  def id: String = _id

  def id_=(value: String): Unit = {
    if (_id.isEmpty) {
      _id = value
    }
  }
}

object Customer3 {
  def main(args: Array[String]): Unit = {
    val bob = new Customer3("0001","bob", "Alvear 1520", "1")
    bob.id = "0001"
    bob.id_$eq("0001")
    bob.id_=("0001")
  }
}

File: training/src/classesandobjects/java/Customer3.java
package classesandobjects.java;

public class Customer3 {
    private String id;
    private final String name;
    private final String address;

    public Customer3(String name, String address) {
        this.name = name;
        this.address = address;
    }

    public void setId(String id) {
        if (this.id.isEmpty()) {
            this.id = id;
        }
    }

    public static void main(String[] args) {
        Customer3 bob = new Customer3("Bob", "Alvear 1520");
        bob.setId("0001");
    }
}

